uaclient/files.py

```diff
@@ -0,0 +1,296 @@
+import json
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Optional
+
+from uaclient import defaults, event_logger, schemas, util
+
+PRIVATE_SUBDIR = "private"
+
+event = event_logger.get_event_logger()
+LOG = logging.getLogger(__name__)
+
+
+class UAFile:
+    _root_directory = defaults.DEFAULT_DATA_DIR
+    _file_name = "file"
```

Comment:
We can define this as a property similar to what we do with `name` for the entitlement classes. This will just guarantee that the subclasses will always define a name